rust/connlib/clients/shared/src/lib.rs

```diff
@@ -1,22 +1,21 @@
 //! Main connlib library for clients.
 pub use connlib_shared::messages::ResourceDescription;
-pub use connlib_shared::{Callbacks, Error};
+pub use connlib_shared::{keypair, Callbacks, Error, LoginUrl, LoginUrlError};
 pub use tracing_appender::non_blocking::WorkerGuard;
 
 use backoff::{backoff::Backoff, ExponentialBackoffBuilder};
-use connlib_shared::control::SecureUrl;
-use connlib_shared::{control::PhoenixChannel, login_url, CallbackErrorFacade, Mode, Result};
+use connlib_shared::StaticSecret;
```

Comment:
> I saw the Dalek docs said it's better to use EphemeralSecret. I assume the reason we use StaticSecret is so we can share the public key with the control plane without having to constantly create and publish new public keys?

We don't use the `StaticSecret` to generate a DH `SharedSecret` but we use the `StaticSecret` both to generate the public key, which we indeed only use once and the bytes to encrypt messages with Wireguard, we can only obtain the underlying bytes with a `StaticSecret`.

Reply:
> I also don't understand how both WebRTC (str0m, right?) and Wireguard (boringtun) are used by connlib.

> Is it that WebRTC is used as a transport layer since STUN/TURN are already standardized for hole-punching, and since it looks more like normal web traffic, and then Wireguard is layered inside for stronger and/or faster crypto than whatever WebRTC uses? (TLS?)

we no longer use WebRTC, we use only the ICE part of `str0m`, which provides us with the transport(hole-punching and all that). `Wireguard` is used to create a tunnel over that, basically encrypting/decrpyting the packets that are sent through the transport. We could use DTLS like webrtc does but it is more complex than Wireguard and we don't need that complexity.

Reply:
> I also don't understand how both WebRTC (`str0m`, right?) and Wireguard (`boringtun`) are used by connlib.
> 
> Is it that WebRTC is used as a transport layer since STUN/TURN are already standardized for hole-punching, and since it looks more like normal web traffic, and then Wireguard is layered inside for stronger and/or faster crypto than whatever WebRTC uses? (TLS?)

We don't use the full webrtc stack, just ICE to figure out a UDP "connection" (two sockets, unencrypted, unauthenticated). Then we layer wireguard on top of that. That is the scope of `snownet`.

The tunnel then adds ACLs to resources on top of that and actually pushes the bytes between device and sockets around :)